launcher/src/test/java/org/apache/brooklyn/launcher/CleanOrphanedLocationsIntegrationTest.java

```diff
@@ -37,54 +38,85 @@
-    private String PERSISTED_STATE_PATH_WITH_MULTIPLE_LOCATIONS_OCCURRENCE = "/orphaned-locations-test-data/fake-multiple-location-for-multiple-search-tests";
-    private String PERSISTED_STATE_PATH_WITHOUT_ORPHANED_LOCATIONS = "/orphaned-locations-test-data/data-without-orphaned-locations";
-    private String PERSISTED_STATE_DESTINATION_PATH = "/orphaned-locations-test-data/copy-persisted-state-destination";
+    private static final Logger LOG = LoggerFactory.getLogger(CleanOrphanedLocationsIntegrationTest.class);
 
+    private final String PERSISTED_STATE_PATH_WITH_ORPHANED_LOCATIONS = "/orphaned-locations-test-data/data-with-orphaned-locations";
+    private final String PERSISTED_STATE_PATH_WITH_MULTIPLE_LOCATIONS_OCCURRENCE = "/orphaned-locations-test-data/fake-multiple-location-for-multiple-search-tests";
+    private final String PERSISTED_STATE_PATH_WITHOUT_ORPHANED_LOCATIONS = "/orphaned-locations-test-data/data-without-orphaned-locations";
 
     private String persistenceDirWithOrphanedLocations;
     private String persistenceDirWithoutOrphanedLocations;
     private String persistenceDirWithMultipleLocationsOccurrence;
-    private String destinationDir;
-    private ManagementContext managementContext;
+    private File destinationDir;
+    private Set<ManagementContext> mgmts;
+    private ManagementContext mgmt;
 
     @BeforeMethod(alwaysRun = true)
-    public void initialize() throws Exception {
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
         persistenceDirWithOrphanedLocations = getClass().getResource(PERSISTED_STATE_PATH_WITH_ORPHANED_LOCATIONS).getFile();
         persistenceDirWithoutOrphanedLocations = getClass().getResource(PERSISTED_STATE_PATH_WITHOUT_ORPHANED_LOCATIONS).getFile();
         persistenceDirWithMultipleLocationsOccurrence = getClass().getResource(PERSISTED_STATE_PATH_WITH_MULTIPLE_LOCATIONS_OCCURRENCE).getFile();
 
-        destinationDir = getClass().getResource(PERSISTED_STATE_DESTINATION_PATH).getFile();
+        destinationDir = Os.newTempDir(getClass());
+        
+        mgmts = Sets.newLinkedHashSet();
     }
 
+    @AfterMethod(alwaysRun = true)
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+        
+        // contents of method copied from BrooklynMgmtUnitTestSupport
+        for (ManagementContext mgmt : mgmts) {
+            try {
+                if (mgmt != null) Entities.destroyAll(mgmt);
+            } catch (Throwable t) {
+                LOG.error("Caught exception in tearDown method", t);
+                // we should fail here, except almost always that masks a primary failure in the test itself,
+                // so it would be extremely unhelpful to do so. if we could check if test has not already failed,
+                // that would be ideal, but i'm not sure if that's possible with TestNG. ?
```

Comment:
TestNG documentation on this: http://testng.org/doc/documentation-main.html#native-dependency-injection